src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImpl.java

```diff
@@ -124,4 +147,129 @@ public void removePaymentAccountAndUserAccountMaps(EntityManager em, List<UserAc
+        }
+
+        return generateUpdatePbaResponse(invalidPbaResponses, pbasFromRequest);
+    }
+
+    private List<PbaUpdateStatusResponse> getInvalidPbaRequests(
+            List<PbaRequest> pbaRequestList, Set<String> pbasFromRequest, String orgId) {
+
+        List<PbaUpdateStatusResponse> invalidPbaResponses = new ArrayList<>();
+
+        pbaRequestList.forEach(pba -> {
+            //check pba is present
+            if (isNullOrEmpty(pba.getPbaNumber())) {
+                invalidPbaResponses.add(generateInvalidResponse("", ERROR_MSG_PBA_MISSING));
+                //check pba is invalid
+            } else if (isPbaInvalid(pba.getPbaNumber())) {
+                invalidPbaResponses.add(generateInvalidResponse(pba.getPbaNumber(), ERROR_MSG_PBA_INVALID_FORMAT));
+                pbasFromRequest.remove(pba.getPbaNumber());
+                //check status is present
+            } else if (isNullOrEmpty(pba.getStatus())) {
+                invalidPbaResponses.add(generateInvalidResponse(pba.getPbaNumber(), ERROR_MSG_STATUS_MISSING));
+                pbasFromRequest.remove(pba.getPbaNumber());
+                //check status is valid
+            } else if (!ACCEPTED.name().equalsIgnoreCase(pba.getStatus())
+                    && !REJECTED.name().equalsIgnoreCase(pba.getStatus())) {
+                invalidPbaResponses.add(generateInvalidResponse(pba.getPbaNumber(), ERROR_MSG_STATUS_INVALID));
+                pbasFromRequest.remove(pba.getPbaNumber());
+            }
+        });
+
+        return invalidPbaResponses;
+    }
+
+    public List<PbaUpdateStatusResponse> acceptOrRejectPbas(
+            List<PaymentAccount> pbasFromDb, List<PbaRequest> pbaRequestList,
+            List<PbaUpdateStatusResponse> invalidPbaResponses, Set<String> pbasFromRequest, String orgId) {
+
+        List<PaymentAccount> pbasToSave = new ArrayList<>();
+        List<PaymentAccount> pbasToDelete = new ArrayList<>();
+
+        if (isNotEmpty(pbasFromDb)) {
+            pbasFromDb.forEach(pba -> pbaRequestList.forEach(pba1 -> {
+                //check PBA belongs to the given Organisation
+                if (!pba.getOrganisation().getOrganisationIdentifier().equals(orgId)) {
+                    invalidPbaResponses.add(generateInvalidResponse(pba.getPbaNumber(), ERROR_MSG_PBA_NOT_IN_ORG));
+                } else {
+                    if (pba1.getPbaNumber().equals(pba.getPbaNumber())) {
+                        //if PBA status is ACCEPTED update status and statusMessage in DB
+                        if (pba1.getStatus().equals(ACCEPTED.name())) {
+                            //check PBA's current status is NOT ACCEPTED before updating to ACCEPTED
+                            if (!ACCEPTED.equals(pba.getPbaStatus())) {
+                                pba.setPbaStatus(ACCEPTED);
+                                pba.setStatusMessage(PBA_STATUS_MESSAGE_ACCEPTED_BY_ADMIN);

```

Comment:
set statusMessage whatever set in request, if not present in request save null

Reply:
Addressed here: https://github.com/hmcts/rd-professional-api/pull/1096/commits/55dcc2cd7029b745a5e8c4a82befb6fb399454f1